Reduce scanner/queue load

tbg opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="35" align="absmiddle" alt="Review on Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/5251)

<!-- Reviewable:end -->

bdarnell commented:
store.go change LGTM, but as discussed offline (and in #5252)  let's remove the gc_queue.go change and instead modify the thresholds on the previous two lines

tschottdorf commented:
Done (and tests fixed), PTAL

bdarnell commented on storage/gc_queue_test.go:
```diff
@@ -91,22 +91,22 @@ func TestGCQueueShouldQueue(t *testing.T) {
+		{bc, 5 * bc * ttl, 10 * ia, 0, now, true, 5 * considerThreshold},
 		// A contribution of 1 from gc, 10/5 from intents.
-		{bc, bc * ttl, 5, 10 * ia, now, true, 1 + 2},
+		{bc, bc * ttl, 5, 10 * ia, now, true, (1 + 2) * considerThreshold},
```
You could move these multiplications down to the point where you compare against `test.priority` so it's not repeated on every line.

tschottdorf replied:
Done.

bdarnell commented:
LGTM

tbg committed:
Improvements to consistency scanner and gc queue

- Set (arbitrary) threshold for GC queue
  Avoid the queue periodically scanning the whole Replica without actually
  getting to delete much. The limit is arbitrary. Fine-tuning ideas welcome.

- Disable MaxIdleTime for the consistency scanner
  This avoids costly recurrent runs of the consistency checker at 5 minute
  intervals.
